qlty-check/src/executor/staging_area.rs

```diff
@@ -218,86 +218,71 @@ impl StagingArea {
-        config_file, to
-    );
-    copy(config_file, to.clone()).with_context(|| {
-        format!(
-            "Failed to copy config file {} to {}",
-            config_file.display(),
-            to.display()
-        )
-    })?;
+    load_config_file_from(config_file, to)
+}
 
-    Ok(())
+pub fn load_config_file_from_source(
+    config_file: impl AsRef<Path>,
+    destination: impl AsRef<Path>,
+) -> Result<String> {
+    load_config_file_from(
+        &config_file,
+        destination
+            .as_ref()
+            .join(config_file.as_ref().file_name().unwrap()),
+    )
 }
 
 pub fn load_config_file_from_qlty_dir(
-    config_file: &Path,
+    config_file: impl AsRef<Path>,
     workspace: &Workspace,
-    destination: &Path,
+    destination: impl AsRef<Path>,
 ) -> Result<String> {
-    let config_file_name = config_file.file_name().unwrap();
+    let config_file_name = config_file.as_ref().file_name().unwrap();
     let from = workspace.library()?.configs_dir().join(config_file_name);
+    let to = destination.as_ref().join(config_file_name);
+    load_config_file_from(from, to)
+}
 
-    if from.exists() {
-        let to = destination.join(config_file_name);
-        if to.exists() {
-            debug!("Config file already exists in workspace: {:?}", to);
-            return Ok(to.display().to_string());
-        }
+fn load_config_file_from(from: impl AsRef<Path>, to: impl AsRef<Path>) -> Result<String> {
+    let from = from.as_ref();
+    let to = to.as_ref();
+    if !from.exists() {
+        return Ok("".to_string());
+    } else if to.exists() {
+        debug!("Config file already exists in workspace: {:?}", to);
+        return Ok(to.display().to_string());
+    }
 
-        debug!(
-            "Symlinking config file from qlty dir: {:?} -> {:?}",
-            from, to
-        );
+    debug!("Symlinking config file: {:?} -> {:?}", from, to);
 
-        let result: std::io::Result<_>;
-        #[cfg(windows)]
-        {
-            result = std::os::windows::fs::symlink_file(from.clone(), to.clone());
-        }
-        #[cfg(unix)]
-        {
-            result = std::os::unix::fs::symlink(from.clone(), to.clone());
-        }
+    symlink(from, to).with_context(|| {
+        format!(
+            "Failed to symlink config file {} to {}",
+            from.display(),
+            to.display()
+        )
+    })?;
 
-        result.with_context(|| {
-            format!(
-                "Failed to symlink config file {} to {}",
-                from.display(),
-                to.display()
-            )
-        })?;
+    Ok(to.display().to_string())
+}
 
-        return Ok(to.display().to_string());
-    }
+#[cfg(windows)]
+fn symlink(from: &Path, to: &Path) -> std::io::Result<()> {
+    std::os::windows::fs::symlink_file(from, to)
+}
 
-    Ok("".to_string())
+#[cfg(unix)]
+fn symlink(from: &Path, to: &Path) -> std::io::Result<()> {
+    std::os::unix::fs::symlink(from, to)
 }
```

Comment:
I think we have this platform-aware symlink stuff in a few cases. Can we extract to a common utility function?

Reply:
I think cleaning up symlink code should probably be a separate PR as it would probably necessitate a separate shared crate.